pkg/component/watch.go

```diff
@@ -39,29 +58,28 @@ func addRecursiveWatch(watcher *fsnotify.Watcher, path string, ignores []string)
 		return nil
 	}
 
+	// If path to be watched is directory, walk the directory recursively ignoring all ignores path and paths under them and accumulate list of all such(eligible) files
 	folders := []string{}
 	err = filepath.Walk(path, func(newPath string, info os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
-
-		if info.IsDir() {
+		// If the src root is watched, automatically all files and folders under it will be watched immaterial of contents of ignores.
+		// So, inorder to avoid the above, src root dir is not watched
```

Comment:
`in order`

Reply:
Updated PR is expected not to have these